src/plugins/data/server/search/routes/search.ts

```diff
@@ -12,82 +12,97 @@ import { reportServerError } from '@kbn/kibana-utils-plugin/server';
-          {
-            legacyHitsTotal: schema.maybe(schema.boolean()),
-            sessionId: schema.maybe(schema.string()),
-            isStored: schema.maybe(schema.boolean()),
-            isRestore: schema.maybe(schema.boolean()),
+export function registerSearchRoute(router: DataPluginRouter): void {
+  router.versioned
+    .post({
+      path: `${SEARCH_API_BASE_URL}/{strategy}/{id?}`,
+      access: 'internal',
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            params: schema.object({
+              strategy: schema.string(),
+              id: schema.maybe(schema.string()),
+            }),
+            body: schema.object(
+              {
+                legacyHitsTotal: schema.maybe(schema.boolean()),
+                sessionId: schema.maybe(schema.string()),
+                isStored: schema.maybe(schema.boolean()),
+                isRestore: schema.maybe(schema.boolean()),
+              },
+              { unknowns: 'allow' }
+            ),
           },
-          { unknowns: 'allow' }
-        ),
+        },
       },
-    },
-    async (context, request, res) => {
-      const {
-        legacyHitsTotal = true,
-        sessionId,
-        isStored,
-        isRestore,
-        ...searchRequest
-      } = request.body;
-      const { strategy, id } = request.params;
-      const abortSignal = getRequestAbortedSignal(request.events.aborted$);
+      async (context, request, res) => {
+        const {
+          legacyHitsTotal = true,
+          sessionId,
+          isStored,
+          isRestore,
+          ...searchRequest
+        } = request.body;
+        const { strategy, id } = request.params;
+        const abortSignal = getRequestAbortedSignal(request.events.aborted$);
 
-      try {
-        const search = await context.search;
-        const response = await search
-          .search(
-            { ...searchRequest, id },
-            {
-              abortSignal,
-              strategy,
-              legacyHitsTotal,
-              sessionId,
-              isStored,
-              isRestore,
-            }
-          )
-          .pipe(first())
-          .toPromise();
+        try {
+          const search = await context.search;
+          const response = await search
+            .search(
+              { ...searchRequest, id },
+              {
+                abortSignal,
+                strategy,
+                legacyHitsTotal,
+                sessionId,
+                isStored,
+                isRestore,
+              }
+            )
+            .pipe(first())
+            .toPromise();
 
-        return res.ok({ body: response });
-      } catch (err) {
-        return reportServerError(res, err);
+          return res.ok({ body: response });
+        } catch (err) {
+          return reportServerError(res, err);
+        }
       }
-    }
-  );
+    );
 
-  router.delete(
-    {
+  router.versioned
+    .delete({
```

Comment:
Consider adding a couple of unit tests for `DELETE` api.

Reply:
:+1: 422bec396f7